### PR TITLE
feat: add job posting schema

### PR DESCRIPTION
**Pull request by sanderdewinter**
### Description

Basic implementation for job posting according to the issue. If this implementation is too specific I might just use raw json definition in `useSchemaOrg` like suggested in [this comment](https://github.com/harlan-zw/unhead-schema-org/issues/10#issuecomment-1370660892)

### Linked Issues
https://github.com/harlan-zw/unhead-schema-org/issues/10

### CODE REVIEW

**Comment by harlan-zw** [comment]
Amazing, great stuff @sanderdewinter. Beyond the comments I've put in the PR, I've pushed up a couple of commits, can you pull them into this pr:
- fixed the snapshot issue causing you to have to update unrelated tests
- moved Place and VirtualLocation to root nodes so you don't have to duplicate them

**Comment by sanderdewinter** [comment]
@harlan-zw Thanks for your help. I think something went wrong since I don't see any comments yet on this PR.

**Comment by harlan-zw** [comment]
Oops, sorry should be visible now

File: packages/unhead-schema-org/src/nodes/JobPosting/index.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import type { ResolvableDate, Thing } from '../../types'
+import { resolvableDateToDate } from '../../utils'
+import { defineSchemaOrgResolver } from '../../core'
+import type { Organization } from '../Organization'
+import type { Place } from './Place'
+import type { MonetaryAmount } from './MonetaryAmount'
+
+export interface JobPostingSimple extends Thing {
```
[the remark this answers is not in the document]
Comment (sanderdewinter) [reply packages/unhead-schema-org/src/nodes/JobPosting/index.ts:11]: I've added those simple types but did not find the identifier field in the schema you suggested

**Review by harlan-zw — COMMENTED** [review]
lgtm

**Comment by harlan-zw** [comment]
Thanks again @sanderdewinter, have merged and released in v0.4.0, and have also bumped nuxt-schema-org and @vueuse/schema-org.

**Pull request merged**